src/services/remoteService.ts

```diff
@@ -194,4 +199,36 @@ export class LocalSSHService extends Disposable implements ILocalSSHService {
             await release();
         }
     }
+
+    async saveRestartInfo() {
+        const connInfo = getGitpodRemoteWindowConnectionInfo(this.context);
+        if (!connInfo) {
+            return;
+        }
+
+        await this.context.globalState.update(`${WORKSPACE_STOPPED_PREFIX}${connInfo.sshDestStr}`, { workspaceId: connInfo.connectionInfo.workspaceId, gitpodHost: connInfo.connectionInfo.gitpodHost, remoteUri: connInfo.remoteUri.toString() } as WorkspaceRestartInfo);
+    }
+
+    checkForStoppedWorkspaces(flow: UserFlowTelemetryProperties) {
+        const keys = this.context.globalState.keys();
+        const stopped_ws_keys = keys.filter(k => k.startsWith(WORKSPACE_STOPPED_PREFIX));
+        for (const k of stopped_ws_keys) {
+            const ws = this.context.globalState.get<WorkspaceRestartInfo>(k)!;
+            this.context.globalState.update(k, undefined);
+            if (new URL(flow.gitpodHost).host === new URL(ws.gitpodHost).host) {
+                this.showWsNotRunningDialog(ws, { ...flow, workspaceId: ws.workspaceId, gitpodHost: ws.gitpodHost });
+            }
+        }
+    }
+
+    private async showWsNotRunningDialog({ workspaceId, gitpodHost, remoteUri }: WorkspaceRestartInfo, flow: UserFlowTelemetryProperties) {
+        const msg = `Workspace ${workspaceId} is not running. Please restart the workspace.`;
```

Comment:
I wonder what can we show instead here? Right now it is not very helpful. workspace id does not provide much contextual information. In browser we have an URL to the original context on the stopping screen and info about dirty changes, so a user can instead how is valuable is this workspace or can be rejected.

I wonder whether maybe we can show webview instead with UI similar to browser but styled with Gitpod theme. It should be another PR though.